beacon-chain/db/kv/state_diff_cache.go

```diff
@@ -0,0 +1,58 @@
+package kv
+
+import (
+	"math"
+	"sync"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/state"
+	"github.com/OffchainLabs/prysm/v6/config/params"
+	"github.com/pkg/errors"
+)
+
+const unset = math.MaxUint64
+
+type stateDiffCache struct {
+	mu      sync.RWMutex
+	anchors map[int]state.ReadOnlyBeaconState
+	offset  uint64
+}
+
+func newStateDiffCache() *stateDiffCache {
+	return &stateDiffCache{
+		anchors: make(map[int]state.ReadOnlyBeaconState, len(params.StateHierarchyExponents())),
```

Comment:
Why are we using a map here if we can access the elements by `anchors[level]` using just a slice? checking for nil on the state should be the same as checking for existence of the map key